firebase-firestore/src/androidTest/java/com/google/firebase/firestore/TransactionTest.java

```diff
@@ -709,23 +709,23 @@ public void testDoesNotRetryOnPermanentError() {
   public void testRetryOnAlreadyExistsError() {
     final FirebaseFirestore firestore = testFirestore();
     DocumentReference doc = firestore.collection("foo").document();
-    CountDownLatch latch = new CountDownLatch(2);
+    AtomicInteger retryCount = new AtomicInteger(0);
     waitFor(
         firestore.runTransaction(
             transaction -> {
-              latch.countDown();
+              retryCount.getAndIncrement();
```

Comment:
Save the result of `retryCount.getAndIncrement()` to a local variable, and use that later on, rather than invoking more methods on `retryCount`, such as `retryCount.get()`.